src/invidious/helpers/youtube_api.cr

```diff
@@ -2,120 +2,200 @@
-    "client" => {
-      "hl"            => "en",
-      "gl"            => region || "US", # Can't be empty!
-      "clientName"    => "WEB",
-      "clientVersion" => HARDCODED_CLIENT_VERS,
-    },
-  }
-end
-
-####################################################################
-# request_youtube_api_browse(continuation)
-# request_youtube_api_browse(browse_id, params, region)
-#
-# Requests the youtubei/v1/browse endpoint with the required headers
-# and POST data in order to get a JSON reply in english that can
-# be easily parsed.
-#
-# The region can be provided, default is US.
-#
-# The requested data can either be:
-#
-#  - A continuation token (ctoken). Depending on this token's
-#    contents, the returned data can be comments, playlist videos,
-#    search results, channel community tab, ...
-#
-#  - A playlist ID (parameters MUST be an empty string)
-#
-def request_youtube_api_browse(continuation : String)
-  # JSON Request data, required by the API
-  data = {
-    "context"      => make_youtube_api_context("US"),
-    "continuation" => continuation,
-  }
-
-  return _youtube_api_post_json("/youtubei/v1/browse", data)
-end
-
-def request_youtube_api_browse(browse_id : String, params : String, region : String = "US")
-  # JSON Request data, required by the API
-  data = {
-    "browseId" => browse_id,
-    "context"  => make_youtube_api_context(region),
-  }
-
-  # Append the additionnal parameters if those were provided
-  # (this is required for channel info, playlist and community, e.g)
-  if params != ""
-    data["params"] = params
+  # Hard-coded constants required by the API
+  HARDCODED_API_KEY     = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8"
+  HARDCODED_CLIENT_VERS = "2.20210330.08.00"
+
+  ####################################################################
+  # make_context(region)
+  #
+  # Return, as a Hash, the "context" data required to request the
+  # youtube API endpoints.
+  #
+  private def make_context(region : String | Nil) : Hash
+    return {
+      "client" => {
+        "hl"            => "en",
+        "gl"            => region || "US", # Can't be empty!
+        "clientName"    => "WEB",
+        "clientVersion" => HARDCODED_CLIENT_VERS,
+      },
+    }
   end
 
-  return _youtube_api_post_json("/youtubei/v1/browse", data)
-end
+  ####################################################################
+  # browse(continuation)
+  # browse(browse_id, params)
+  # browse(browse_id, params, region)
+  #
+  # Requests the youtubei/v1/browse endpoint with the required headers
+  # and POST data in order to get a JSON reply in english that can
+  # be easily parsed.
+  #
+  # A region can be provided, default is US.
+  #
+  # The requested data can either be:
+  #
+  #  - A continuation token (ctoken). Depending on this token's
+  #    contents, the returned data can be playlist videos, channel
+  #    community tab content, channel info, ...
+  #
+  #  - A playlist ID (parameters MUST be an empty string)
+  #
```

Comment:
[Crystal has a standard documentation syntax](https://crystal-lang.org/reference/syntax_and_semantics/documenting_code.html); might be a good idea to switch to that.

Reply:
I am following those, with a few exceptions in order to make the code more readable (e.g the `###...` header to visually split the different sections, or the `browse(continuation)` type of lines).

Reply:
And they honestly help a lot when reading the code directly! However, the main reason why those documentation guidelines exists is for the `crystal docs` tool, which really doesn't handle the extra header very well the last time I checked.

So  basically, this come down to if we'd like to support the standard `crystal docs` tool or not.

Reply:
Glad to hear it! Given that invidious is not meant to be a library for something else, I don't think that using `crystal docs` is required. The code has to be documented, but only for future maintainers/contributors.